images/fcos-bastion-image/root/usr/bin/clean-registries.sh

```diff
@@ -6,6 +6,11 @@ for port in "${ports[@]}"; do
   disk_use=$(df /opt/registry-"${port}" --output='pcent' | grep -o '[0-9]*')
   if [ "$disk_use" -gt 85 ]; then
     rm -rf /opt/registry-"${port}"/data/docker/registry/v2/repositories/*
+    sed -i -e '/readonly/{n;s/enabled:.*/enabled: true/;}' /opt/registry-"${port}"/config.yaml
+    systemctl restart registry@"${port}".service
+    sleep 5
```

Comment:
I think we'd have a function that continuously check and sleep 0.5s until the registry container is healthy/ready.

Reply:
Testing after adding function that waits for container to be running:

core@dhcp-1-235-141:~$ sudo journalctl -u clean-registries.service -l --no-pager -o cat
Starting clean-registries.service - Service to clean up mirrored container images cache...
+ mapfile -t ports
++ systemctl list-units 'registry@*' --no-pager --quiet
++ awk '-F[@.]' '{print $2}'
+ for port in "${ports[@]}"
++ df /opt/registry-5000 --output=pcent
++ grep -o '[0-9]*'
+ disk_use=21
+ '[' 21 -gt 11 ']'
+ rm -rf '/opt/registry-5000/data/docker/registry/v2/repositories/*'
+ sed -i -e '/readonly/{n;s/enabled:.*/enabled: true/;}' /opt/registry-5000/config.yaml
+ systemctl restart registry@5000.service
+ wait_for_container_running registry-5000
+ local container_name=registry-5000
+ echo 'Waiting for container registry-5000 to become running...'
Waiting for container registry-5000 to become running...
+ podman inspect -f '{{.State.Running}}' registry-5000
+ sleep 1
+ podman inspect -f '{{.State.Running}}' registry-5000
+ echo 'Container registry-5000 is now running.'
Container registry-5000 is now running.
+ podman exec registry-5000 /bin/registry garbage-collect /etc/docker/registry/config.yml
2024-02-27 22:07:25.580176213 +0000 UTC m=+0.051822732 container exec 6147193aea69ead9ece4b224a014de14bdd6b4775b0352bead4a523a75d1b1df (image=quay.io/libpod/registry:2.8.2, name=registry-5000, PODMAN_SYSTEMD_UNIT=registry@5000.service)
0 blobs marked, 0 blobs and 0 manifests eligible for deletion
2024-02-27 22:07:25.61543736 +0000 UTC m=+0.087083879 container exec_died 6147193aea69ead9ece4b224a014de14bdd6b4775b0352bead4a523a75d1b1df (image=quay.io/libpod/registry:2.8.2, name=registry-5000, PODMAN_SYSTEMD_UNIT=registry@5000.service)
+ sed -i -e '/readonly/{n;s/enabled:.*/enabled: false/;}' /opt/registry-5000/config.yaml
+ systemctl restart registry@5000.service
+ for port in "${ports[@]}"
++ grep -o '[0-9]*'
++ df /opt/registry-6001 --output=pcent
+ disk_use=21
+ '[' 21 -gt 11 ']'
+ rm -rf '/opt/registry-6001/data/docker/registry/v2/repositories/*'
+ sed -i -e '/readonly/{n;s/enabled:.*/enabled: true/;}' /opt/registry-6001/config.yaml
+ systemctl restart registry@6001.service
+ wait_for_container_running registry-6001
+ local container_name=registry-6001
+ echo 'Waiting for container registry-6001 to become running...'
Waiting for container registry-6001 to become running...
+ podman inspect -f '{{.State.Running}}' registry-6001
+ sleep 1
+ podman inspect -f '{{.State.Running}}' registry-6001
+ echo 'Container registry-6001 is now running.'
Container registry-6001 is now running.
+ podman exec registry-6001 /bin/registry garbage-collect /etc/docker/registry/config.yml
2024-02-27 22:07:27.537199429 +0000 UTC m=+0.052035462 container exec b55b756937425a28211e98d54647752db74087e72860b4c51ea90aabc29764a5 (image=quay.io/libpod/registry:2.8.2, name=registry-6001, PODMAN_SYSTEMD_UNIT=registry@6001.service)
0 blobs marked, 0 blobs and 0 manifests eligible for deletion
2024-02-27 22:07:27.57809303 +0000 UTC m=+0.092929043 container exec_died b55b756937425a28211e98d54647752db74087e72860b4c51ea90aabc29764a5 (image=quay.io/libpod/registry:2.8.2, name=registry-6001, PODMAN_SYSTEMD_UNIT=registry@6001.service)
+ sed -i -e '/readonly/{n;s/enabled:.*/enabled: false/;}' /opt/registry-6001/config.yaml
+ systemctl restart registry@6001.service
+ for port in "${ports[@]}"
++ df /opt/registry-6002 --output=pcent
++ grep -o '[0-9]*'
+ disk_use=21
+ '[' 21 -gt 11 ']'
+ rm -rf '/opt/registry-6002/data/docker/registry/v2/repositories/*'
+ sed -i -e '/readonly/{n;s/enabled:.*/enabled: true/;}' /opt/registry-6002/config.yaml
+ systemctl restart registry@6002.service
+ wait_for_container_running registry-6002
+ local container_name=registry-6002
+ echo 'Waiting for container registry-6002 to become running...'
Waiting for container registry-6002 to become running...
+ podman inspect -f '{{.State.Running}}' registry-6002
+ sleep 1
+ podman inspect -f '{{.State.Running}}' registry-6002
+ echo 'Container registry-6002 is now running.'
Container registry-6002 is now running.
+ podman exec registry-6002 /bin/registry garbage-collect /etc/docker/registry/config.yml
2024-02-27 22:07:29.623937974 +0000 UTC m=+0.051583599 container exec 680630eef32fc7021cbc30e6297c98a72d062f52d6fcefb72f6bd343394686d4 (image=quay.io/libpod/registry:2.8.2, name=registry-6002, PODMAN_SYSTEMD_UNIT=registry@6002.service)
0 blobs marked, 0 blobs and 0 manifests eligible for deletion
2024-02-27 22:07:29.664089795 +0000 UTC m=+0.091735430 container exec_died 680630eef32fc7021cbc30e6297c98a72d062f52d6fcefb72f6bd343394686d4 (image=quay.io/libpod/registry:2.8.2, name=registry-6002, PODMAN_SYSTEMD_UNIT=registry@6002.service)
+ sed -i -e '/readonly/{n;s/enabled:.*/enabled: false/;}' /opt/registry-6002/config.yaml
+ systemctl restart registry@6002.service
+ exit 0